Client/Ecosia/UI/WhatsNew/DataProvider/WhatsNewJSONDataProvider.swift

```diff
@@ -0,0 +1,38 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Core
+
+struct JSONDataProvider: DataProvider {
+    
+    let jsonData: Data?
+    private struct WhatsNewItemRoot: Decodable {
+        let items: [RemoteWhatsNewItem]
+    }
+
+    private struct RemoteWhatsNewItem: Decodable {
+        let imageURL: URL
+        let title: String
+        let subtitle: String
+    }
+
+    func fetchData() throws -> [WhatsNewItem] {
```

Comment:
Thanks for the comment @lucaschifino and the explicit no blocker for that 💪
For sure `fetchData()` might cause some misalignments in what truly does 👍 . It works mostly as an adapter component as the data will be already fetched async via another function from outside this context.+

I'm fine with removing the code, no probs at all as it will be a separate commit that could potentially be reverted.

When it comes to Braze integration we have two options as far I know/remember. 
Braze could give us a _"notification"_ telling us to do something alongside a payload of info. We might be able to receive the notification at any given/defined time. The _notification payload_ will contain all the info to show the view (which essentially will be the `Data` this provider will receive.

The other option (the one I liked the most and proposed upfront) will be to delegate to Braze the design of the entire view (this is possible via their direct SDK integration). Of course, this is now paused. Thus, the programmatically implemented WhatsNew page.

I'm inclined to leave the `DataProvider` just to better separate the concerns, as the ViewModel already contains the `items`  for a given `DataProvider` (even if just local for now).

What I'm trying to build up here, mostly, is a sort of defined, standardised and testable architecture that would make our view creation no brainer when it comes of what pattern to utilize and/or decision to make to show things like "a list".

Reply:
Thanks for the explanation 🙏 Braze integration delegating the design sounds indeed promising 🙌 

It's ok for me if you prefer to leave the `DataProvider` then, if you want you can maybe rethink the `fetchData` naming to avoid confusions, but it is not a must.

Reply:
Updated it into `getData`. Hope this clarifies it better its intents